### ISSUE TITLE
Use ormolu style for default

### ISSUE THREAD

**Issue by dpwiz** [fourmolu/fourmolu#128]
To make fourmolu easier to adopt revert default style to match ormolu.

The binary then can be a drop-in replacement and all the features/differences can be explained one by one, against a reference background.

Since it is possible (mandatory?) to support ormolu style anyway, this will make option testing more sensible (see comments at #49) and test cases can be used to document every option visually.

This will make a move from "ormolu" to "ormolu with tweaks", not a lateral "some opinionated formatter, to _another_ opinionated formatter with tweaks".

**Comment by brandonchinn178** [comment]
Makes sense to me! We format the Fourmolu code with the `fourmolu.yaml` in the repo, which should match Ormolu's style, intentionally (https://github.com/fourmolu/fourmolu/blob/master/DEVELOPER.md#contributing). So I think a good plan would be to:
1. Make the `defaultPrinterOpts` the same as the currently checked in `fourmolu.yaml`
2. Remove `fourmolu.yaml` (to ensure that formatting Fourmolu source code itself uses Ormolu style)
3. Change the tests to use `defaultPrinterOpts` for ormolu tests and the current-defaults of Fourmolu for Fourmolu tests (related: https://github.com/fourmolu/fourmolu/issues/49)

**Comment by brandonchinn178** [comment]
Two more benefits of doing this:
* In light of the new testing strategy in #49, using ormolu for defaults means we can get rid of testing the default fourmolu config (i.e. delete all the `*-four-out.hs` files + all the changes we had to make to PrinterSpec.hs)
* Make it easier for people, when writing issues, to check what Ormolu's behavior is, and seeing if their issue is actually an issue with fourmolu or if its also present in ormolu

**Comment by istathar** [comment]
This would be terrible. We rely on _fourmolu_ using the current **formoulu** project defaults and do not need to override with a config file to get away from **ormolou**'s craziness. It would be awful to wake up one day and find all the formatting change out from under us.

**Comment by georgefst** [comment]
> This would be terrible. We rely on _fourmolu_ using the current **formoulu** project defaults and do not need to override with a config file to get away from **ormolou**'s craziness. It would be awful to wake up one day and find all the formatting change out from under us.

This is pretty much my thinking as well. I think some of Ormolu's formatting decisions are just fundamentally misguided, and I've argued this in various threads in the past, particularly those concerned with deciding on Fourmolu's default behaviour.

Besides, I think it's just too late to change. I suspect a lot of people are using Fourmolu with default settings, and regressing to Fourmolu's behaviour would wind them up immensely.

(I've been meaning for a while to respond here, but I haven't been able to find a very diplomatic way of phrasing my concerns.)

**Comment by dpwiz** [comment]
We have a `fourmolu` fork of our own that we are struggling to maintain due to amount of churn imposed by this weird double-upstream situation. Basically I want everything to converge so the maintenance will be easier.

I don't know if sticking to upstream defaults will help (it certainly wouldn't in the short term) but such a concession may make the situation less polarized. As it is now, the situation looks pretty damning.

Maybe I'm barking up the wrong tree and a correct resolution would be extraction of a common core to a 3rd library with pooled maintenance effort and without stepping on each others' toes.

**Comment by brandonchinn178** [comment]
@dpwiz whats the purpose of your fork? if you just add a config file, you can workaround this behavior

**Comment by dpwiz** [comment]
It's a bunch of pull requests to support our code style that we are waiting to be merged.

Adding insult to injury, there's a HLS fork too that wraps this patched 16molu.
